IBM/openapi-validator
# Error responses

Purpose:
- Error responses should be robust and give the user enough information to resolve their problems. The added validation recommends certain fields be included and recommends a schema for those fields.

Changes:
- Add a custom Spectral rule that applies a custom Spectral function to validate 4xx and 5xx responses.
- Add a custom Spectral rule that validates the error response schema is an object, has fields, such as `trace`, `code`, `errors` or `error`, etc. and validates the schema of those fields to ensure the error response follows good practices.
- Defines a deduplication function and passes it to spectral to avoid deduplicating `response-error-response-schema` errors with the same path.
- Add a custom Spectral rule to check for `content` object in request bodies and non-204 responses.

Tests:
- Add tests to ensure warnings are issued for missing fields or invalid/incomplete schemas for each field.
- Add negative tests to ensure no warnings for valid error response.
- Add tests to ensure warning when content object not provided

Docs:
- document `reponse-error-response-schema` rule
- document `content-entry-provided` rule
